app/src/popup.js

```diff
@@ -57,153 +53,133 @@ function getUnitCode({lesson}) {
-
-        var currentTab = tabs[0].url;
-        var domain = currentTab.match(/^[\w-]+:\/{2,}\[?([\w\.:-]+)\]?(?::[0-9]*)?/)[1];
-        _gaq.push(['_trackEvent', domain, 'visited']);
-
-        const {primaryFiles} = lesson.video.media.media.current;
-        if (downloadHD) {
-          const {s3Url, width, height} = primaryFiles[1];
-          // TODO: URL for access outside of Australia.
-          return `https://${domain}/media/download?s3Url=` + s3Url + "&fileName=hd1.mp4&resolution=" + width.toString() + "x" + height.toString();
-        } else {
-          const {s3Url, width, height} = primaryFiles[0];
-          return `https://${domain}/media/download?s3Url=` + s3Url + "&fileName=sd1.mp4&resolution=" + width.toString() + "x" + height.toString();
-        }
-      });
-}
-
 // Job of this function is to listen init mediaLessons once per click.
-function webRequestOnComplete(xhrRequest) {
+async function webRequestOnComplete(xhrRequest) {
   console.log("Media Lessons obtained!");
   _gaq.push(['_trackEvent', 'webReqFunc', 'loaded']);
   if (mediaLessons === undefined) {
     mediaLessons = xhrRequest;
     // Now perform the request again ourselves and download files.
-    var getMediaLessonsRequest = new Request(mediaLessons.url, {method: 'GET'});
-    fetch(
-      getMediaLessonsRequest,
-      {
-        method: 'GET',
-            credentials: 'include'
-      })
-      .then((getMediaLessonsResponse) => getMediaLessonsResponse.json())
-      .then((getMediaLessonsJson) => {
-        console.log(getMediaLessonsJson);
-        downloadables = getMediaLessonsJson.data.filter((dataItem) => {
-          return canDownload(dataItem);
-        });
-
-        // sort downloadables
-        downloadables.sort((a, b) => {
-          const nameA = getVideoFileName(a), nameB = getVideoFileName(b);
-          if (nameA < nameB) return -1;
-          else if (nameA == nameB) return 0;
-          else return 1;
-        });
-        const lectureTable = document.getElementById("lectures");
-        const lectureSelect = document.getElementById("lectureSelect");
-        downloadables.forEach((downloadable) => {
-          const option = document.createElement("option");
-          option.defaultSelected = true;
-          const name = getUnitCode(downloadable) + "_" + getVideoFileName(downloadable);
-
-          option.innerHTML = name;
-          lectureSelect.appendChild(option);
-        });
-
-        var downloadButton = document.getElementById('download');
-          downloadButton.disabled = false;
-        });
-  }
-}
+    const getMediaLessonsRequest = new Request(mediaLessons.url, { method: 'GET', credentials: 'include' });
+    const getMediaLessonsResponse = await fetch(getMediaLessonsRequest);
+    const getMediaLessonsJson = await getMediaLessonsResponse.json();
+
+    const courseDataRequest = new Request('https://echo360.org.uk/user/enrollments', { method: 'GET', credentials: 'include' });
```

Comment:
It'd be nice if this could support other domains too, similiar to @ericjiang97's previous comment.

Reply:
Also curious why the endpoint is 'enrollments' - feel like its either a typo either here or on echo360's end.

Reply:
I think it's on their end, as I can successfully download videos now (after changing the domains).

Reply:
The use of enrolments here is to build a better filename based on course name